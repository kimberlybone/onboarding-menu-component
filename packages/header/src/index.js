export { MainTitle } from './components/MainTitle';
export { Status } from './components/Status';
