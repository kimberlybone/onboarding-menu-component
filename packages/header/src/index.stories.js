import React                              from 'react';
import { storiesOf, addDecorator }        from '@storybook/react';
import styled                             from "styled-components";
import { withKnobs, select, text }				from '@storybook/addon-knobs';
import { MainTitle, Status }              from './index';
storiesOf( 'Header', module )
  .addDecorator(withKnobs)
  .add('MainTitle', () => {
    return <MainTitle></MainTitle>
  })
  .add('Status', () => {
    return <Status></Status>
  })
