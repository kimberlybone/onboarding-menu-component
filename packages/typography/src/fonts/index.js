import WebFont from 'webfontloader';

const WebFontConfig = {
  custom: {
    families: ['GothamNarrowUltra', 'GothamNarrowLight'],
    urls: ['http://www.tightventures.com/fonts/export.css']
  },
  google:{
  	families: ['Open Sans']
  }
};

WebFont.load(WebFontConfig);

export const Fonts = {
	GothamNarrowUltra: 'GothamNarrowUltra',
	GothamNarrowLight: 'GothamNarrowLight',
  OpenSans: 'Open Sans'
};
