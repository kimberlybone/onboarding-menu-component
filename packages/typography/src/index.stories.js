import React                            from 'react';
import { storiesOf, addDecorator }      from '@storybook/react';
import { withKnobs, text, select }      from '@storybook/addon-knobs';
import { Jumbo, 
         SubJumbo, 
         Heading, 
         SubHeading, 
         Body, 
         TitleHighlight }               from './index';
import { Colors }                       from '@narrow/theme';

const primitiveOptions = {
  'h1': 'h1',
  'h2': 'h2',
  'h3': 'h3',
  'h4': 'h4',
  'h5': 'h5',
  'h6': 'h6',
  'p' : 'p',
  'span': 'span'
};

const colorOptions = {
  GrayVeryDark: Colors.GrayVeryDark,
  Blue: Colors.Blue,
  Orange: Colors.Orange,
  Purple: Colors.Purple,
  Green: Colors.Green,
  Red: Colors.Red,
  Yellow: Colors.Yellow,
  GrayDark: Colors.GrayDark,
  Gray: Colors.Gray,
  GrayLight: Colors.GrayLight,
  White: Colors.White
};

const sizeOptions = {
  ExtraLarge: 'extraLarge',
  Large: 'large',
  Medium: 'medium',
  Small: 'small',
  ExtraSmall: 'extraSmall'
};

storiesOf('Typography', module)
  .addDecorator(withKnobs)
  .add('Jumbo', () => {
    const size = select('Size', sizeOptions, 'extraLarge');
    return <Jumbo
    {...{[size]: true}}
    color={select('Color', colorOptions)}
    primitive={select('Primitive', primitiveOptions)}
    >
      {text('Title', 'Jumbo')}
    </Jumbo>;

  })
  .add('Heading', () => {
    const size = select('Size', sizeOptions, 'extraLarge');
    return <Heading
    {...{[size]: true}}
    color={select('Color', colorOptions)}
    primitive={select('Primitive', primitiveOptions)}
    >
      {text('Title', 'Heading')}
    </Heading>;

  })
  .add('Body', () => {
    const size = select('Size', sizeOptions, 'extraLarge');
    return <Body
    {...{[size]: true}}
    color={select('Color', colorOptions)}
    primitive={select('Primitive', primitiveOptions)}
    >
      {text('Title', 'Body')}
    </Body>;

  })
  .add('TitleHighlight', () => {
    const size = select('Size', sizeOptions, 'extraLarge');
    return <TitleHighlight
    {...{[size]: true}}
    color={select('Color', colorOptions)}
    lineColor={select('Line Color', colorOptions)}>
      {text('Title', 'Title')}
    </TitleHighlight>;

  });