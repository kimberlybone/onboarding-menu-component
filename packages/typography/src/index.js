import { Fonts } from './fonts';
import { Jumbo } from './components/Jumbo';
import { Heading } from './components/Heading';
import { Body } from './components/Body';
import { TitleHighlight } from './components/TitleHighlight';
import Text  from './components/Text';

export { Fonts, Jumbo, Heading, Body, Text, TitleHighlight };