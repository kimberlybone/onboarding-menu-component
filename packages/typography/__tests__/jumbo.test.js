import React        from 'react';
import renderer     from 'react-test-renderer';
import { Jumbo }         from '../src';
import { Colors }         from '@narrow/theme';
import 'jest-styled-components'

jest.useFakeTimers();

let testFn = jest.fn();

//Snapshot
test('Jumbo renders correctly', () => {
  const tree = renderer.create(<Jumbo title={'Jumbo'}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

//Default props
test('renders default props correctly', () => {
  const testRenderer = renderer.create(<Jumbo 
                                        title={'Jumbo'}/>);

  const testInstance = testRenderer.root;

  expect(testInstance.props.color).toBe(Colors.GrayVeryDark);
  expect(testInstance.props.primitive).toBe('h1');
});


//Other Prop Tests

test('renders fourExtraSmall correctly', () => {
  const testRenderer = renderer.create(<Jumbo title={'Jumbo fourExtraSmall'} 
                                              fourExtraSmall/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.fourExtraSmall).toBeTruthy();
});

test('renders threeExtraSmall correctly', () => {
  const testRenderer = renderer.create(<Jumbo title={'Jumbo threeExtraSmall'} 
                                              threeExtraSmall/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.threeExtraSmall).toBeTruthy();
});

test('renders twoExtraSmall correctly', () => {
  const testRenderer = renderer.create(<Jumbo title={'Jumbo twoExtraSmall'} 
                                              twoExtraSmall/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.twoExtraSmall).toBeTruthy();
});

test('renders extraSmall correctly', () => {
  const testRenderer = renderer.create(<Jumbo title={'Jumbo extraSmall'} 
                                              extraSmall/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.extraSmall).toBeTruthy();
});

test('renders small correctly', () => {
  const testRenderer = renderer.create(<Jumbo title={'Jumbo small'} 
                                              small/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.small).toBeTruthy();
});

test('renders medium correctly', () => {
  const testRenderer = renderer.create(<Jumbo title={'Jumbo medium'} 
                                              medium/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.medium).toBeTruthy();
});

test('renders large correctly', () => {
  const testRenderer = renderer.create(<Jumbo title={'Jumbo large'} 
                                              large/>);
  const testInstance = testRenderer.root; 
  expect(testInstance.props.large).toBeTruthy();
});

test('renders extraLarge correctly', () => {
  const testRenderer = renderer.create(<Jumbo title={'Jumbo extraLarge'} 
                                              extraLarge/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.extraLarge).toBeTruthy();
});

test('renders twoExtraLarge correctly', () => {
  const testRenderer = renderer.create(<Jumbo title={'Jumbo twoExtraLarge'} 
                                              twoExtraLarge/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.twoExtraLarge).toBeTruthy();
});

test('renders threeExtraLarge correctly', () => {
  const testRenderer = renderer.create(<Jumbo title={'Jumbo threeExtraLarge'} 
                                              threeExtraLarge/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.threeExtraLarge).toBeTruthy();
});

//// Config Tests
test('renders JUMBO fontconfig Large correctly', () => {
  const tree = renderer.create(<Jumbo title={'JUMBO fontconfig Large'} large />).toJSON();
  expect(tree).toHaveStyleRule('font-family', `GothamNarrowUltra`);
  expect(tree).toHaveStyleRule('font-weight', `bold`);
  expect(tree).toHaveStyleRule('font-size', `104px`);
  expect(tree).toHaveStyleRule('line-height', `104px`);
});

//// Viewport Tests
test('renders smallMobile correctly', () => {
  const testRenderer = renderer.create(<Jumbo viewport = {'smallMobile'}/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.viewport).toBe('smallMobile');
});
test('renders mobile correctly', () => {
  const testRenderer = renderer.create(<Jumbo viewport = {'mobile'}/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.viewport).toBe('mobile');
});
test('renders tablet correctly', () => {
  const testRenderer = renderer.create(<Jumbo viewport = {'tablet'}/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.viewport).toBe('tablet');
});
test('renders smallDesktop correctly', () => {
  const testRenderer = renderer.create(<Jumbo viewport = {'smallDesktop'}/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.viewport).toBe('smallDesktop');
});
test('renders desktop correctly', () => {
  const testRenderer = renderer.create(<Jumbo viewport = {'desktop'}/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.viewport).toBe('desktop');
});
test('renders largeDesktop correctly', () => {
  const testRenderer = renderer.create(<Jumbo viewport = {'largeDesktop'}/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.viewport).toBe('largeDesktop');
});
test('renders hdDesktop correctly', () => {
  const testRenderer = renderer.create(<Jumbo viewport = {'hdDesktop'}/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.viewport).toBe('hdDesktop');
});
test('renders uhdDesktop correctly', () => {
  const testRenderer = renderer.create(<Jumbo viewport = {'uhdDesktop'}/>);
  const testInstance = testRenderer.root;
  expect(testInstance.props.viewport).toBe('uhdDesktop');
});
