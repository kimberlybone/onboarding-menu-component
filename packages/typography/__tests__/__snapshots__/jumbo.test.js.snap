// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jumbo renders correctly 1`] = `
.c0 {
  color: #282828;
  font-size: 120px;
  line-height: 120px;
  margin: 0;
}

.c1 {
  font-weight: bold;
  font-family: GothamNarrowUltra;
  color: #282828;
  -webkit-letter-spacing: 0.02em;
  -moz-letter-spacing: 0.02em;
  -ms-letter-spacing: 0.02em;
  letter-spacing: 0.02em;
  text-transform: uppercase;
}

<h1
  className="c0 c1"
  color="#282828"
  fontSize={120}
  title="Jumbo"
/>
`;
